src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/ChannelResource.java

```diff
@@ -314,11 +314,8 @@ public ResponseEntity<Void> registerUsersToChannel(@PathVariable Long courseId,
         checkEntityIdMatchesPathIds(channelFromDatabase, Optional.of(courseId), Optional.of(channelId));
         var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
         channelAuthorizationService.isAllowedToRegisterUsersToChannel(channelFromDatabase, usersLoginsToRegister, requestingUser);
-        Set<User> usersToRegister = new HashSet<>();
-        usersToRegister.addAll(conversationService.findUsersInDatabase(course, addAllStudents, addAllTutors, addAllInstructors));
-        usersToRegister.addAll(conversationService.findUsersInDatabase(usersLoginsToRegister.stream().toList()));
-        conversationService.registerUsersToConversation(course, usersToRegister, channelFromDatabase, Optional.empty());
-        usersToRegister.forEach(user -> singleUserNotificationService.notifyClientAboutConversationCreationOrDeletion(channelFromDatabase, user, requestingUser,
+        Set<User> registeredUsers = channelService.registerUsersToChannel(addAllStudents, addAllTutors, addAllInstructors, usersLoginsToRegister, course, channelFromDatabase);
+        registeredUsers.forEach(user -> singleUserNotificationService.notifyClientAboutConversationCreationOrDeletion(channelFromDatabase, user, requestingUser,
```

Comment:
As far as I can see this is sent for each default channel. Can we only create one notification telling the user that they were added to all default channels (maybe mentioning them in there)?

Reply:
The notifications are not sent when channels are created automatically at the moment. This notification here is sent only when a user is added manually.